modules/extensions/src/collide/collide.ts

```diff
@@ -0,0 +1,92 @@
+import {Accessor, Layer, LayerContext, LayerExtension} from '@deck.gl/core';
+import collide from './shader-module';
+import CollideEffect from './collide-effect';
+
+const defaultProps = {
+  getCollidePriority: {type: 'accessor', value: 0},
+  collideEnabled: true,
+  collideGroup: {type: 'string', value: 'default'},
+  collideTestProps: {}
+};
+
+export type CollideExtensionProps<DataT = any> = {
+  /**
+   * Accessor for collision priority. Must return a number in the range -1000 -> 1000. Features with higher values are shown preferentially.
+   */
+  getCollidePriority?: Accessor<DataT, number>;
+
+  /**
+   * Enable/disable collisions. If collisions are disabled, all objects are rendered.
+   * @default true
+   */
+  collideEnabled: boolean;
+
+  /**
+   * Collision group this layer belongs to. If it is not set, the 'default' collision group is used
+   */
+  collideGroup?: string;
+
+  /**
+   * Props to override when rendering collision map
+   */
+  collideTestProps?: {};
+};
+
+/** Allows layers to hide overlapping objects. */
+export default class CollideExtension extends LayerExtension {
+  static defaultProps = defaultProps;
+  static extensionName = 'CollideExtension';
+
+  isEnabled(layer: Layer<CollideExtensionProps>): boolean {
+    return layer.getAttributeManager() !== null;
+  }
+
+  getShaders(this: Layer<CollideExtensionProps>): any {
+    return {modules: [collide]};
+  }
+
+  /* eslint-disable camelcase */
+  draw(this: Layer<CollideExtensionProps>, {uniforms, context, moduleParameters}: any) {
+    const {collideEnabled, collideGroup} = this.props;
+    const {drawToCollideMap, collideFBOs = {}} = moduleParameters;
+    const collideGroups = Object.keys(collideFBOs);
+    const enabled = collideEnabled && collideGroup && collideGroups.includes(collideGroup);
+    uniforms.collide_enabled = enabled;
+
+    if (drawToCollideMap) {
```

Comment:
I recommend moving this logic to `shaderModule.getUniforms`. The MaskExtension was calculating some of the uniforms in `draw` because it needed the layer's projection methods.